azure-kusto-data/azure/kusto/data/helpers.py

```diff
@@ -48,9 +48,13 @@ def dataframe_from_result_table(table: "Union[KustoResultTable, KustoStreamingRe
 
     columns = [col.column_name for col in table.columns]
     frame = pd.DataFrame(table.raw_rows, columns=columns)
+    print(frame.dtypes)
+    print(frame.infer_objects().dtypes)
 
     # fix types
     for col in table.columns:
+        if col.column_type == "string" and hasattr(pd, "StringDType"):
```

Comment:
Suggest adding a check on the return value as well, just in case.

Reply:
On what? the return value of the type? not sure how?

Reply:
I think once `hasattr(pd, "StringDType")` is `True`, you can check its value via `pd.StringDType`. It isn't critical, but this can help ensure the type is not only defined on the `df`, but as expected, which could theoretically happen as the external package evolves.

(I'd expect the value to be `string`, but I could be wrong.)

Reply:
It's not a string, the type *is* pd.StringDType, so I can really only compare it to itself..